client/replays/browse-local-replays.tsx

```diff
@@ -19,6 +41,244 @@ async function getReplayFolderPath() {
+
+const ReplayPanelContainer = styled.div<{ $isLoading: boolean }>`
+  width: 100%;
+  padding: 16px;
+  background-color: ${background400};
+
+  opacity: ${props => (props.$isLoading ? 0.6 : 1)};
+  transition: opacity linear 100ms;
+`
+
+const InfoContainer = styled.div`
+  display: grid;
+  grid-auto-flow: row;
+  grid-auto-rows: max-content;
+  grid-template-columns: repeat(8, 1fr);
+  grid-gap: 24px 24px;
+`
+
+const PlayerListContainer = styled.div`
+  grid-column: 1 / 6;
+`
+
+const TeamLabel = styled.div`
+  ${overline};
+  ${singleLine};
+
+  height: 24px;
+  line-height: 24px;
+  margin: 0 8px;
+
+  color: ${colorTextSecondary};
+`
+
+const PlayerContainer = styled.div`
+  width: 100%;
+  height: 56px;
+  padding: 8px;
+
+  display: flex;
+  align-items: center;
+  text-align: left;
+
+  & + ${TeamLabel} {
+    margin-top: 16px;
+  }
+`
+
+const RaceRoot = styled.div`
+  position: relative;
+  width: auto;
+  height: 32px;
+`
+
+const StyledRaceIcon = styled(RaceIcon)`
+  width: auto;
+  height: 100%;
+`
+
+const PlayerName = styled.div`
+  ${headline6};
+  ${singleLine};
+  margin-left: 16px;
+  margin-right: 8px;
+  flex-grow: 1;
+`
+
+const ReplayInfoContainer = styled.div`
+  grid-column: 6 / 9;
+  height: auto;
+
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: flex-start;
+`
+
+const StyledMapThumbnail = styled(MapThumbnail)`
+  ${shadow2dp};
+`
+
+const MapName = styled.div`
+  ${headline6};
+  ${singleLine};
+  margin-top: 8px;
+`
+
+const StyledTooltip = styled(Tooltip)`
+  width: 100%;
+  display: flex;
+  justify-content: center;
+`
+
+const ReplayInfoText = styled.div`
+  ${subtitle1};
+  ${singleLine};
+  margin: 8px 0;
+
+  color: ${colorTextSecondary};
+`
+
+const MapNoImageContainer = styled.div`
+  position: relative;
+  width: 100%;
+  height: auto;
+  border-radius: 2px;
+  contain: content;
+`
+
+const ReplayActionsContainer = styled.div`
+  display: flex;
+  justify-content: flex-end;
+  align-items: center;
+  width: 100%;
+  height: 56px;
+`
+
+export function ReplayExpansionPanel({ file }: ExpansionPanelProps) {
+  const dispatch = useAppDispatch()
+  const [replayHeader, setReplayHeader] = useState<ReplayHeader>()
+  const [replayShieldBatteryData, setReplayShieldBatteryData] = useState<ReplayShieldBatteryData>()
+  const gameId = replayShieldBatteryData?.gameId
+  const replayUserIds = replayShieldBatteryData?.userIds
+  const gameInfo = useAppSelector(s => s.games.byId.get(gameId ?? ''))
+  const mapInfo = useAppSelector(s => s.maps2.byId.get(gameInfo?.mapId ?? ''))
+  const usersById = useAppSelector(s => s.users.byId)
+
+  const mapNameRef = useRef<HTMLDivElement>(null)
+  const isMapNameCutOff = useCutoffElement(mapNameRef.current)
+  const gameTypeRef = useRef<HTMLDivElement>(null)
+  const isGameTypeCutOff = useCutoffElement(gameTypeRef.current)
+
+  useEffect(() => {
+    getReplayHeader(file.path).then(header => setReplayHeader(header))
```

Comment:
You need to handle errors on both of these, no Promises with unattached error handlers